File: January/Python/Insert Delete GetRandom O(1).py
class RandomizedSet:

    def __init__(self):
        self.random_dict = {}

    def insert(self, val: int) -> bool:
        if val not in self.random_dict:
            self.random_dict[val] = True
            return True
        else:
            return False

    def remove(self, val: int) -> bool:
        if val in self.random_dict:
            del self.random_dict[val]
            return True
        else:
            return False

    def getRandom(self) -> int:
        return list(self.random_dict.keys())[random.randint(0,len(self.random_dict)-1)]


# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()

File: January/Java/Insert Delete GetRandom O(1).java
class RandomizedSet {
    Set<Integer> s;
    public RandomizedSet() {
        s=new HashSet<>();
    }
    
    public boolean insert(int val) {
        if(s.contains(val)){
            return false;
        }else{
            s.add(val);
            return true;
        }
    }
    
    public boolean remove(int val) {
        if(s.contains(val)){
            s.remove(val);
            return true;
        }else{
            return false;
        }
    }
    
    public int getRandom() {
        List<Integer> list=new ArrayList<Integer>(s);
        int r=(int)(Math.random()*s.size());

        return list.get(r);
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet obj = new RandomizedSet();
 * boolean param_1 = obj.insert(val);
 * boolean param_2 = obj.remove(val);
 * int param_3 = obj.getRandom();
 */
